controllers/componentdetectionquery_controller.go

```diff
@@ -126,6 +129,15 @@ func (r *ComponentDetectionQueryReconciler) Reconcile(ctx context.Context, req c
 		if context == "" {
 			context = "./"
 		}
+		if source.Revision == "" {
+			log.Info(fmt.Sprintf("Look for default branch of repo %s... %v", source.URL, req.NamespacedName))
+			source.Revision, err = github.GetDefaultBranchFromRepo(source.URL, r.GitHubClient, ctx)
```

Comment:
what happens when we have a CDQ

```
apiVersion: appstudio.redhat.com/v1alpha1
kind: ComponentDetectionQuery
metadata:
  name: componentdetectionquery-sample
spec:
  git:
    url: https://github.com/devfile-samples/devfile-sample-java-springboot-basic
    devfileUrl: https://raw.githubusercontent.com/maysunfaisal/devfileexample/main/case2/devfile.yaml
```

we are explicitly specifying a devfile url

Reply:
What's the usecase for this exactly? I remember this came up a week or two ago and wondered why we would need to specify a devfile during a CDQ?

Reply:
it will go with this path: https://github.com/redhat-appstudio/application-service/blob/d5ebec0fda3b9dfcebc36f65cae2ba6508611702/controllers/componentdetectionquery_controller.go#L231-L240
in this case , unrelated to the revision.